scala/private/common.bzl

```diff
@@ -166,12 +166,33 @@ def _provider_of_dependency_label_of(dependency, path):
 
 # TODO this seems to have limited value now that JavaInfo has everything
 def create_java_provider(scalaattr, transitive_compile_time_jars):
-    return java_common.create_provider(
-        use_ijar = False,
-        compile_time_jars = scalaattr.compile_jars,
-        runtime_jars = scalaattr.transitive_runtime_jars,
-        transitive_compile_time_jars = depset(
-            transitive = [transitive_compile_time_jars, scalaattr.compile_jars],
-        ),
-        transitive_runtime_jars = scalaattr.transitive_runtime_jars,
-    )
+    compile_time_deps = [
+        JavaInfo(
+            output_jar = jar,
+            compile_jar = jar,
+        )
+        #TODO: Avoid flattening the depset here.
+        for jar in transitive_compile_time_jars.to_list()
```

Comment:
isn't this putting all transitive dependencies as direct dependencies? Note previously there was a distinction between `compile_time_jars` and `transitive_compile_time_jars`. It seems we are losing that distinction here.

Reply:
@johnynek I don't think so since @iirina later uses `compile_time_deps` as dependencies of `runtime_providers` which means you have another layer of indirection.
If I understand correctly this preserves semantics for you but breaks it for us (who use +1).
Maybe what's needed is to create another indirection provider and put that one as a dependency on the final provider. This is hacky but should be simple to do and maintain the current semantics.